ratatui-crossterm/Cargo.toml

```diff
@@ -14,11 +14,31 @@ exclude.workspace = true
 edition.workspace = true
 rust-version.workspace = true
 
+[package.metadata.docs.rs]
+features = ["document-features"]
+rustdoc-args = ["--cfg", "docsrs"]
+
 [features]
-default = ["underline-color"]
+default = ["underline-color", "crossterm_0_29"]
+
+#! One of the following versions of crossterm must be enabled. By default, the latest version is
+#! enabled. This will always default to the most recent version of crossterm, but you can override
+#! this by specifying the version you want to use in your Cargo.toml file. The purpose of these
+#! features is to allow widget libraries to depend on specific versions of crossterm in a way that
+#! doesn't cause version conflicts. A compiler error will be raised if you try to use multiple
+#! versions of crossterm at the same time.
+
+## Enables crossterm 0.28.x
+crossterm_0_28 = ["dep:crossterm_0_28"]
+## Enables crossterm 0.29.x
+crossterm_0_29 = ["dep:crossterm_0_29"]
```

Comment:
It might get a bit annoying to add a new feature flag for the each new version of crossterm, even the patches. Or are you thinking of only doing this for the minor versions?

Are we fine with that? If they decide to speed up their release cycle we might end up with so many feature flags. Maybe we can let the upstream know that we are doing this and hear what they think?

Reply:
It's only necessary for semver incompatible versions (0.x -> 0.x+1). Patch version doesn't matter.
It's also fairly unlikely that crossterm will speed up development IMO. Either way, we can roll with it as needed.
Perhaps we support last 2 versions or something like that. The rationale here is to make it possible for downstream widget crates to support several versions as needed.

Reply:
Yeah, supporting only the last 2-3 versions sound reasonable to me. We should then update these docs accordingly (we only support blah blah)

I still believe we should adjust this PR a bit to make it easier to add new versions (see my other comments)

Reply:
done